### ISSUE TITLE
jetstreamExotic requiring secure connection (no opt-out)

### ISSUE THREAD

**Issue by diogobaeder** [argoproj/argo-events#2938]
**Describe the bug**
Argo Events requires secure connection to use jetstreamExotic, but it shouldn't - not because secure connections shouldn't be used, but because there are cases when they're not required, like on dev environments (which is my case) and when running inside a service mesh that provides mTLS.

**To Reproduce**
1. Run a NATS server with JetStream enabled, but without any secure connections enabled
2. Spin up Argo Events
3. Create an EventBus pointing to the NATS server, using jetstreamExotic
4. Check the logs

**Expected behavior**
I should be able to connect the EventBus with the external NATS server without problems.

**Screenshots**
(None)

**Environment (please complete the following information):**
 - Kubernetes: v1.27.3
 - Argo: v3.5.1
 - Argo Events: latest+ec96ecd

**Additional context**
Here's where the secure connection is forced: https://github.com/argoproj/argo-events/blob/master/eventbus/jetstream/base/jetstream.go#L70

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by diogobaeder** [comment]
Yeah, it looks like it's not very active at the moment. I don't know if this has anything to do with Octopus buying Codefresh, but I sure hope the project gets some love again.

For the time being, though, I was forced to stop using it.

**Comment by amarkdotdev** [comment]
Opened https://github.com/argoproj/argo-events/pull/4114 — when TLS is unset we connect over plain TCP instead of forcing nats.Secure.
